File: Models/Pickup.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace splat.Models
{
    [Index(nameof(PickupStatus), nameof(StudentInfo))]
    public class Pickup
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public double? Weight { get; set; }
        [Required]
        public PickupStatus PickupStatus { get; set; }
        [DataType(DataType.DateTime)]
        public DateTime? PickupTime { get; set; }
        [DataType(DataType.DateTime)]
        public DateTime? CanceledTime { get; set; }
        [Required]
        [Column(TypeName = "jsonb")]
        public Student StudentInfo { get; set; }
        [DataType(DataType.DateTime)]
        public DateTime SubmittedAt { get; set; }
        [Column(TypeName = "jsonb")]
        public HouseholdInfo? HouseholdInfo { get; set; }
        [Required]
        [Column(TypeName = "jsonb")]
        public ItemRequest[] ItemRequests { get; set; }
        [DataType(DataType.DateTime)]
        public DateTime RequestedPickupTime { get; set; }
        public string OtherNotes { get; set; }
    }
    public enum PickupStatus
    {
        PENDING,
        WAITING,
        DISBURSED,
        CANCELED
    }
}

File: splat/ClientApp/src/models/Pickup.ts
import type { ItemRequest } from './ItemRequest';
import type { StudentInfo } from './StudentInfo';
import type { HouseholdInfo } from './HouseholdInfo';

export type Pickup = {
    id: string | null;
    weight?: number;
    pickupStatus: PickupStatus;
    pickupTime?: Date;
    canceledTime?: Date;
    submittedAt: Date | null;
    studentInfo: StudentInfo;
    householdInfo?: HouseholdInfo;
    itemRequests: ItemRequest[];
    requestedPickupTime: Date;
    otherNotes: string;
};

export enum PickupStatus {
    PENDING = 0,
    WAITING = 1,
    DISBURSED = 2,
    CANCELED = 3,
};
